### PR TITLE
nv2a: Add --enable-renderdoc build config

### PR DESCRIPTION
**Pull request by abaire**
Followup from #674

### CODE REVIEW

File: hw/xbox/nv2a/debug.h [hunk 1]
```diff
@@ -37,14 +37,12 @@
 #endif
 
 // #define DEBUG_NV2A_REG
-// #define DEBUG_NV2A_GL
+ #define DEBUG_NV2A_GL
```
Comment (mborgerson) [inline hw/xbox/nv2a/debug.h]: Should not be enabled by default

Comment (abaire) [reply hw/xbox/nv2a/debug.h]: Missed that after testing, sorry!

File: meson.build [hunk 2]
```diff
@@ -1224,6 +1224,22 @@ endif
...
+if have_system and get_option('renderdoc').enabled()
+  if not cc.has_header('renderdoc_app.h')
+    error('renderdoc required, but renderdoc_app.h was not found')
+  else
+    if targetos == 'linux'
+       if not libdl.found()
+         error('renderdoc required, but libdl was not found')
+         renderdoc = not_found
+       else
+         renderdoc = declare_dependency(link_args: '-ldl')
```
Comment (mborgerson) [inline meson.build]: We already did the libdl detection above, so to conditionally link libdl it's something like:

`specific_ss.add(when: 'CONFIG_RENDERDOC', if_true: [libdl])`

Comment (abaire) [reply meson.build]: Done (I think)

**Comment by mborgerson** [comment]
Thanks

**Pull request merged**